pr-build.yml

```diff
@@ -0,0 +1,76 @@
+name: PR Build
+
+on:
+  pull_request:
+    branches: [master]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: technote-space/get-diff-action@v1
+        id: git_diff_content
+        with:
+          SUFFIX_FILTER: .html
+          PREFIX_FILTER: content/files/
+
+      - name: Display Git diffs
+        run: |
+          echo ${{ steps.git_diff_content.outputs.diff }}
+
+      - name: Setup Node.js environment
+        uses: actions/setup-node@v2-beta
+        with:
+          node-version: "12"
+      - name: Display Node & Yarn version
+        run: |
+          node --version
+          yarn --version
+
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+      - uses: actions/cache@v1
+        id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+
+      - name: Install all yarn packages
+        run: yarn --frozen-lockfile
+
+      # TODO figure out a way to set up a cache that we can put things
+      # like _all-content.json into.
+
+      - name: Build changed content
+        if: steps.git_diff_content.outputs.diff
```

Comment:
This is an extremely important line. It basically nullifies the whole thing if there's nothing in the `git diff` (with the prefix `content/files/` and suffix `.html`). 
So if you make a PR that only edits the `package.json` (and `yarn.lock`) or some `client/src/styles.scss` then it won't build this workflow at all. 

What we COULD do is always build some. If there are `content/files/**` in the git diff, build those, otherwise, build a handful of favorites we hardcode into the test. 

Which reminds me, if this `if: ` statement decides to *not* build any pages, then the somewhat expensive `yarn install --frozen` (on line 45) would still run. I'll fix that.